File: app/src/main/java/edu/gatech/cs2340/spacetrader/views/RegisterActivity.java
package edu.gatech.cs2340.spacetrader.views;

import android.app.ProgressDialog;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import edu.gatech.cs2340.spacetrader.R;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;

import edu.gatech.cs2340.spacetrader.views.MainActivity;
/**
 * Register activity
 *
 * @author Group 46B NO MAC
 * @version 1.0
 */
@SuppressWarnings({"LongLine", "ChainedMethodCall"})
public class RegisterActivity extends AppCompatActivity {

    private EditText inputEmail;
    private EditText inputPassword;
    private FirebaseAuth auth;
    private ProgressDialog PD;


    /**
     * lets user register to play game
     * @param savedInstanceState Bundle savedInstanceState
     */
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_register);

        PD = new ProgressDialog(this);
        PD.setMessage("Loading...");
        PD.setCancelable(true);
        PD.setCanceledOnTouchOutside(false);

        auth = FirebaseAuth.getInstance();

        if (auth.getCurrentUser() != null) {
            startActivity(new Intent(RegisterActivity.this, MainActivity.class));
            finish();
        }

        inputEmail = findViewById(R.id.email_edit_text);
        inputPassword = findViewById(R.id.password_edit_text);
        Button btnSignUp = findViewById(R.id.register_button);
        Button btnLogin = findViewById(R.id.back_button);

        btnSignUp.setOnClickListener(new View.OnClickListener() {
            @Override            public void onClick(View view) {
                final String email = inputEmail.getText().toString();
                final String password = inputPassword.getText().toString();

                try {
                    if ((!password.isEmpty()) && (!email.isEmpty())) {
                        PD.show();
                        auth.createUserWithEmailAndPassword(email, password)
                                .addOnCompleteListener(RegisterActivity.this, new OnCompleteListener<AuthResult>() {
                                    @Override
                                    public void onComplete(@NonNull Task<AuthResult> task) {
                                        if (!task.isSuccessful()) {
                                            Toast.makeText(
                                                    RegisterActivity.this,
                                                    "Authentication Failed",
                                                    Toast.LENGTH_LONG).show();
                                            Log.v("error", task.getResult().toString());
                                        } else {
                                            Intent intent = new Intent(RegisterActivity.this, MainActivity.class);
                                            startActivity(intent);
                                            finish();
                                        }
                                        PD.dismiss();
                                    }
                                });
                    } else {
                        Toast.makeText(
                                RegisterActivity.this,
                                "Fill All Fields",
                                Toast.LENGTH_LONG).show();
                    }
                } catch (Exception e) {
                    e.printStackTrace();
                }
            }
        });

        btnLogin.setOnClickListener(new View.OnClickListener() {
            @Override            public void onClick(View view) {
                finish();
            }
        });


    }
}

File: app/src/main/java/edu/gatech/cs2340/spacetrader/views/RegisterActivity.kt
package edu.gatech.cs2340.spacetrader.views

import android.content.Context
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.os.PersistableBundle
import android.util.Log
import android.view.View
import android.widget.*

import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.*
import edu.gatech.cs2340.spacetrader.models.user
import edu.gatech.cs2340.spacetrader.R
import edu.gatech.cs2340.spacetrader.WelcomeActivity

class RegisterActivity : AppCompatActivity() {
    private lateinit var userName: EditText
    private lateinit var password: EditText
    private lateinit var register: Button
    private lateinit var back: Button
    private lateinit var user: user
    private var mAuth: FirebaseAuth? = null
    private var mDatabase: DatabaseReference? = null
    private var mContext: Context? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_register)
        mContext = this
        mDatabase = FirebaseDatabase.getInstance().reference
        mAuth = FirebaseAuth.getInstance()
        userName = findViewById<View>(R.id.email_edit_text) as EditText
        password = findViewById<View>(R.id.password_edit_text) as EditText
        register = findViewById<View>(R.id.register_button) as Button
        back = findViewById<View>(R.id.back_button) as Button

        register.setOnClickListener {
            user = user(userName.text.toString(), password.text.toString())
            if (userName.length() == 0 || password.length() == 0) {
                Toast.makeText(this, "Please Enter text in email/password", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }
            if (user.checkPasswordStrength() <= 1) {
                Toast.makeText(this, "Please Enter a stronger password", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            mAuth!!.createUserWithEmailAndPassword(user.username, user.password).addOnCompleteListener {
                if (!it.isSuccessful) {
                    Toast.makeText(applicationContext, (it.getException())?.message , Toast.LENGTH_LONG).show()
                    return@addOnCompleteListener

                }
                // else if successful

                Log.d("main", "Successfully created user with uid: $(it.user.id)")
                val userId = mAuth!!.currentUser!!.uid
                user = user(userName.text.toString(), password.text.toString())
                FirebaseDatabase.getInstance().reference.child("users").child(userId).setValue(user)
            }.addOnFailureListener {

            }

            startActivity(Intent(this, MainActivity:: class.java))

            back.setOnClickListener {
                startActivity(Intent(this, WelcomeActivity:: class.java ))
            }


        }
    }
}